plugins/destination/mssql/client/migrate.go

```diff
@@ -59,26 +59,36 @@ func (c *Client) schemaTables(ctx context.Context, tables schema.Tables) (schema
 	return schemaTables, nil
 }
 
-func (c *Client) normalizedTables(tables schema.Tables) schema.Tables {
-	var normalized schema.Tables
-	for _, table := range tables.FlattenTables() {
-		table := table
-		for i := range table.Columns {
-			// Since multiple schema types can map to the same MSSQL type we need to normalize them to avoid false positives when detecting schema changes
-			// This should never return an error
-			typ, _ := queries.SchemaType(table.Name, table.Columns[i].Name, queries.SQLType(table.Columns[i].Type))
-			table.Columns[i].Type = typ
-		}
-		// If there are no PKs, we use CqID as PK
-		pks := table.PrimaryKeys()
-		if !c.pkEnabled() || len(pks) == 0 {
-			table.Columns.Get(schema.CqIDColumn.Name).CreationOptions.PrimaryKey = true
-		}
-
-		normalized = append(normalized, table)
+func (c *Client) normalizeTable(table *schema.Table) {
+	// this is for backward compatibility as I believe this is as of right now defined on the source
+	if c.pkEnabled() && len(table.PrimaryKeys()) == 0 {
```

Comment:
There's an issue with cq ID for cases where pk isn't enabled (append mode).

basically, you need to move this check after the loop and then do the update in the following case:
```go
if !c.pkEnabled() || len(table.PrimaryKeys()) == 0
```
this way you'll add PK to the cq ID if it's append mode or for overwrite modes if no PK defined

Reply:
I don't believe that's correct (since I just copied the code from postgresql, which is our 'canonical destination').
https://github.com/cloudquery/cloudquery/blob/901472697dbd6370079659d0f08d8ef01865b987/plugins/destination/postgresql/client/migrate.go#L162

I believe that the correct behaviour is not to create primary keys at all for append mode (we don't have them for postgresql 'append' mode :) ).
re: "make _cq_id be PK if a PK is not defined" - that's handled by the source according to convo with YP - that was just a backward-compatibility hack that's six months outdated by now :) 

[above is corroborated by https://github.com/cloudquery/cloudquery/blob/901472697dbd6370079659d0f08d8ef01865b987/plugins/destination/postgresql/client/migrate.go#L361-L369]

Reply:
See the second commit in this PR 'don't create PK in append mode' that should fix the failing tests :)